### ISSUE TITLE
Bug in comparing pre-releases.

### ISSUE THREAD

**Issue by sammarx** [jlindsey/semantic#12]
From the semver website - 

"Numeric identifiers always have lower precedence than non-numeric identifiers. A larger set of pre-release fields has a higher precedence than a smaller set, if all of the preceding identifiers are equal. Example: 1.0.0-alpha < 1.0.0-alpha.1 < 1.0.0-alpha.beta < 1.0.0-beta < 1.0.0-beta.2 < 1.0.0-beta.11 < 1.0.0-rc.1 < 1.0.0."

With the semantic gem -

[1] pry(main)> require 'semantic'
=> true
[2] pry(main)> require 'semantic/core_ext'
=> true
[3] pry(main)> '1.0.0-beta.2'.to_version < '1.0.0-beta.11'.to_version
=> false

If you agree this is a bug, I'll take a shot at a PR.

**Comment by perlun** [comment]
Are you sure this is with the **semantic** gem and not the **semver** gem? Your email seems to indicate the latter... :smile:

**Comment by sammarx** [comment]
Whoops.  This is with the semantic gem.  That was a typo.  I'll edit it.

**Comment by dhempy** [comment]
It appears that prerelease is always compared as a string.  For example, it thinks that `1.0.0-99` is greater than `1.0.0-100`
```
require 'semantic'
require 'semantic/core_ext'

def test_ver a, b
  ver_a = a.to_version
  ver_b = b.to_version

  if ver_a < ver_b
    puts "#{ver_a} is less than #{ver_b}"
  elsif ver_a > ver_b
    puts "#{ver_a} is greater than #{ver_b}"
  elsif ver_a == ver_b
    puts "#{ver_a} is equal to #{ver_b}"
  end

  puts "#{a} ->  #{ver_a.inspect}"
  puts "#{b} ->  #{ver_b.inspect}"
end

a = '1.0.0-99'
b = '1.0.0-100'

test_ver(a,b)
```

Running this yields:
```
vagrant@vagrant-ubuntu-trusty-64$ ruby tmp/notes/ver_prerelease.rb
1.0.0-99 is greater than 1.0.0-100
1.0.0-99 ->  #<Semantic::Version:0x00000001ed5e80 @pre="99", @major=1, @minor=0, @patch=0>
1.0.0-100 ->  #<Semantic::Version:0x00000001ed5ae8 @pre="100", @major=1, @minor=0, @patch=0>

vagrant@vagrant-ubuntu-trusty-64$ gem list sem*

*** LOCAL GEMS ***

activerecord-sqlserver-adapter (4.2.3)
awesome_nested_set (3.0.1)
database_cleaner (1.3.0)
devise (3.4.1, 3.2.4)
devise-i18n (0.11.4)
parser (2.2.2.6, 2.2.0.3)
semantic (1.4.1)
sendgrid (1.2.0)
vagrant@vagrant-ubuntu-trusty-64:$
```

**Comment by perlun** [comment]
@dhempy - agree. This is indeed a bug. Would you care to add a failing test case and submit a PR? If you're really eager, you could even try fixing it (but just adding a failing test case is a really good start).

**Comment by ant31** [comment]
#17 is fixing this

**Comment by jlindsey** [comment]
Should be fixed with PR #17 and release 1.5.0.
